src/helper/SM4T_translate.py

```diff
@@ -20,25 +16,59 @@ def check_language_supported_guard(language):
         )
 
 
-def translate_text(text, from_code, to_code):
-    """
-    Translates the given text from the origin language to the target language.
+class Translator:
+    def __init__(self, config: dict):
+        self.device = (
+            f"cuda:{config['device_index']}"
+            if config["translation_device"] == "cuda"
+            else config["translation_device"]
+        )
+        self.tokenizer = SeamlessM4TTokenizer.from_pretrained(
+            config["translation_model"],
+            cache_dir=CONFIG["model_path"],
```

Comment:
Does the caching work properly now? :thinking:

Reply:
unfortunately not :/
and I still have no clue why